atlp-rwanda/e-comm-team-brogrammers-bn
# #184581338 Admin should be able to disable an account for various reasons

#### What does this PR do?
* The PR would likely be adding the feature for an admin to disable a user account and send an email to the user with reasons for the block.
#### Description of Task to be completed?
* The task is to add a feature that allows an admin to disable a user account for various reasons such as fraud, abuse, malicious use or violation of terms of service. The admin should be able to trigger an action to block usage of a particular account and an email should be sent to the user with reasons for the block. Sample violation reasons are provided such as merchants selling illegal items, harassment of other users of the application, and requests from law enforcement.
#### How should this be manually tested?
* Create an admin account and multiple user accounts on the system.
* As the admin, log in to the system and navigate to the user account that needs to be disabled.
* Trigger the action to block usage of the user account.
* Check that the user account has been disabled and the user has received an email with the reasons for the block.
*  Verify that the user is unable to log in to the system.
* PATCH/users/disable/{userId}
#### Any background context you want to provide?
* N/A
#### What are the relevant pivotal tracker/Trello stories?
* [#184581338](https://www.pivotaltracker.com/n/projects/2632830/stories/184581338)

#### Screenshots (if appropriate)
* 
![Screenshot from 2023-03-20 23-12-17](https://user-images.githubusercontent.com/101427855/226466962-9601c2ab-d4d5-435e-97b2-327dc4baed7b.png)
* 
![Screenshot from 2023-03-21 22-09-42](https://user-images.githubusercontent.com/101427855/226716125-275093a8-2231-42c4-bb2c-70a8a537464b.png)



#### Questions:
N/A
